src/main/kotlin/Serialization.kt

```diff
@@ -1,17 +1,5 @@
-import kotlinx.serialization.KSerializer
-import kotlinx.serialization.json.Json
+import kotlinx.serialization.Serializable
 
-
-class Serialization {
-    private val json = Json { ignoreUnknownKeys = true }
-
-
-    fun parseBlockResponse(json: String): BlockResponse {
-        return json.decodeFromString(BlockResponse.serializer())
-    }
-
-
-    private fun String.decodeFromString(serializer: KSerializer<BlockResponse>): BlockResponse {
-        return json.decodeFromString(serializer, this)
-    }
-}
+@Serializable
+data class BlockIds(val id: String)
+data class TxIds(val id: String)
```

Comment:
This should be `@Serializable` as well:

```suggestion
@Serializable
data class TxIds(val id: String)
```

Overall, I believe that the issue is that the classes you're defining simply don't match the content `mempool.space` sends you, so it's normal that deserialization doesn't work...you need to create these `data class` to match what `mempool.space` sends if you want it to work.